src/apps/Settings/Settings.js

```diff
@@ -50,28 +53,37 @@ class Settings extends React.Component {
     walletNetwork: PropTypes.string.isRequired,
     walletWeb3: PropTypes.object.isRequired,
   }
-  static defaultProps = {
-    account: '',
-  }
   state = {
-    defaultEthNode,
-    ipfsGateway: ipfsDefaultConf.gateway,
     currencies: AVAILABLE_CURRENCIES,
+    ethNode: defaultEthNode,
+    ipfsGateway: ipfsDefaultConf.gateway,
     selectedCurrency: filterCurrency(getSelectedCurrency()),
+    selectedNodeInvalid: false,
   }
   handleSelectedCurrencyChange = (index, currencies) => {
     setSelectedCurrency(currencies[index])
     this.setState({ selectedCurrency: currencies[index] })
   }
   handleDefaultEthNodeChange = event => {
-    this.setState({ defaultEthNode: event.target.value })
+    this.setState({
+      ethNode: event.target.value && event.target.value.trim(),
+      selectedNodeInvalid: false,
+    })
   }
   handleIpfsGatewayChange = event => {
-    this.setState({ ipfsGateway: event.target.value })
+    this.setState({
+      ipfsGateway: event.target.value && event.target.value.trim(),
+    })
   }
-  handleNodeSettingsSave = () => {
-    const { defaultEthNode, ipfsGateway } = this.state
-    setDefaultEthNode(defaultEthNode)
+  handleNodeSettingsSave = async () => {
+    const { ethNode, ipfsGateway } = this.state
+
+    if (!(await isValidEthNode(ethNode, network.type))) {
+      this.setState({ selectedNodeInvalid: true })
+      return
```

Comment:
Wondering if we should throw an error here or let the user know something went wrong.

Reply:
Yes! We need to create guidelines for this type of warnings messages and list all current use cases. cc @jounih

Reply:
<img width="575" alt="screen shot 2019-02-15 at 2 38 22 pm" src="https://user-images.githubusercontent.com/4166642/52860310-6046fe00-312f-11e9-98bd-5c2b27b77528.png">

This is what the error currently looks like if you give it a bad URI (somewhat similar to some versions of our form validation).